src/metabase/upload/parsing.clj

```diff
@@ -135,7 +165,7 @@
 (defmethod upload-type->parser :metabase.upload/datetime
   [_]
   (comp
-   parse-datetime
+   parse-as-datetime
```

Comment:
Note this needs to be `parse-as-datetime` instead of `parse-datetime` now because `parse-datetime` now doesn't parse date strings like "2022-01-01", but `parse-as-datetime` does.